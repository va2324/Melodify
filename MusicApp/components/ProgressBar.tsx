import React from "react";
import { View, StyleSheet } from "react-native";

type Props = {
  progress: number;
};

export default function ProgressBar({ progress }: Props) {
  return (
    <View style={styles.container}>
      <View style={[styles.progressBar, { width: `${progress}%` }]} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: 200,
    height: 16,
    backgroundColor: "#E5E7EB", 
    borderRadius: 8,
    overflow: "hidden",
  },
  progressBar: {
    height: "100%",
    backgroundColor: "#5543A5",
  },
});
