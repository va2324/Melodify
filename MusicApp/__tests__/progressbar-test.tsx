import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from '@/components/ProgressBar';

describe('progress bar', () => {
    test('renders successfully', () => {
        const tree = renderer.create(<ProgressBar progress={100} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});