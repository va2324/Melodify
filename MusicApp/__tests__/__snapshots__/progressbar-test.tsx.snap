// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress bar renders successfully 1`] = `
<View
  style={
    {
      "backgroundColor": "#E5E7EB",
      "borderRadius": 8,
      "height": 16,
      "overflow": "hidden",
      "width": 200,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#a12916",
          "height": "100%",
        },
        {
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
